index.html

```diff
@@ -687,7 +659,64 @@ <h4><dfn>SmartCardGetStatusChangeOptions</dfn> dictionary</h4>
       </section>
       <section>
         <h3><dfn>connect()</dfn> method</h3>
-        <div class="issue">Write an algorithm for this method.</div>
+        <p>The {{SmartCardContext/connect(readerName, accessMode,
+        preferredProtocols)}} method steps are:</p>
+        <ol>
+          <li>Let |promise:Promise| be [=a new promise=].</li>
+          <li>If [=this=].{{SmartCardContext/[[operationInProgress]]}} is
+            `true`, [=reject=] |promise| with a "{{InvalidStateError}}"
+            {{DOMException}} and return |promise|.</li>
+          <li>Set [=this=].{{SmartCardContext/[[operationInProgress]]}} to
+            `true`.</li>
+          <li>Run the following steps [=in parallel=]:
+            <ol>
+              <li>Let |accessFlags:DWORD| be a [[PCSC5]] `DWORD`
+              [=SmartCardAccessMode/corresponding=] to
+              |accessMode:SmartCardAccessMode|.</li>
+              <li>Let |protocolFlags:DWORD| be a `DWORD` set to `0`.</li>
+              <li>If |preferredProtocols:sequence&lt;SmartCardProtocol&gt;| was
```

Comment:
For extensibility, move _preferredProtocols_ into an options dictionary. This will be useful if, for example, we want to add an `AbortSignal` parameter for PC/SC implementations which support aborting a connection.

Reply:
Done.